dmg/compress.c

```diff
@@ -8,6 +8,11 @@
 #ifdef HAVE_LIBLZMA
   #include <lzma.h>
 
+  #ifndef LZMA_FAIL_FAST
+    // Introduced in 5.3.3
+    #define LZMA_FAIL_FAST 0
```

Comment:
AFAICT, you're setting this to 0 to avoid attempting to pass it (it looks like doing so will result in an `LZMA_OPTION_ERROR` in older versions according to https://tukaani.org/xz/liblzma-api/container_8h.html#aa1f469ed3d4b2eaf12f8081657efc9d5). This is a bit confusing because in this block, it has a different value than it would have from the library. Can we move the #ifdef onto the function using it? Or perhaps have indirection on what we pass? Eg:
```
#ifdef LZMA_FAIL_FAST
  #define DECODE_OPTS LZMA_FAIL_FAST
#
  #define DECODE_OPTS 0
#ENDIF
```

Reply:
Sure, that makes sense.